modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/80_script_score.yml

```diff
@@ -546,3 +546,46 @@
   - match: {hits.hits.0.highlight.company.0: "<em>ABC</em> company"}
   - match: {hits.hits.1.highlight.company.0: "<em>ABC</em> <em>ABCD</em> company"}
   - match: {hits.hits.2.highlight.company.0: "<em>ABCD</em> company"}
+
+---
+# This test guards against a subtle edge case in the fetch phase related to source-loading and
```

Comment:
This protects against reintroducing the bug in #31000. I should have added this test when closing the bug, but better late than never!